File: app/src/main/java/jgappsandgames/smartreminderslite/date/WeekActivity.java
package jgappsandgames.smartreminderslite.date;

// Java
import java.util.Calendar;

// Android OS
import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;

// Views
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.BaseAdapter;
import android.widget.Button;
import android.widget.ListView;
import android.widget.Toast;

// App
import jgappsandgames.smartreminderslite.R;

import jgappsandgames.smartreminderslite.holder.TaskFolderHolder.OnTaskChangedListener;
import jgappsandgames.smartreminderslite.home.FirstRun;

// Save
import jgappsandgames.smartreminderssave.date.DateManager;
import jgappsandgames.smartreminderssave.settings.Settings;
import jgappsandgames.smartreminderssave.tags.TagManager;
import jgappsandgames.smartreminderssave.tasks.TaskManager;
import jgappsandgames.smartreminderssave.utility.FileUtility;

/**
 * WeekActivity
 * Created by joshua on 10/9/17.
 */
public class WeekActivity extends Activity implements OnClickListener, OnTaskChangedListener {
    // Data
    private int week_active;

    // Views
    private ListView tasks;
    private Button previous;
    private Button next;

    // Adapters
    private BaseAdapter adapter;

    // LifeCycle Methods
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // Set Content View
        setContentView(R.layout.activity_date);

        // First Run
        FileUtility.loadFilePaths(this);
        if (FileUtility.isFirstRun()) {
            Intent first_run = new Intent(this, FirstRun.class);
            startActivity(first_run);
        }

        // Load Data
        Settings.load();
        TaskManager.load();
        TagManager.load();

        week_active = 0;

        // Set Title
        setTitle();

        // Find Views
        tasks = findViewById(R.id.tasks);
        previous = findViewById(R.id.previous);
        next = findViewById(R.id.next);

        // Set Click Listeners
        previous.setOnClickListener(this);
        next.setOnClickListener(this);
    }

    @Override
    protected void onResume() {
        super.onResume();

        adapter = new WeekAdapter(this, week_active);
        tasks.setAdapter(adapter);
    }

    @Override
    protected void onPause() {
        super.onPause();

        save();
    }

    // Menu
    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_auxilary, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.save:
                TaskManager.save();
                Toast.makeText(this, "Saved.", Toast.LENGTH_LONG).show();
                break;

            case R.id.close:
                finish();
        }

        return super.onOptionsItemSelected(item);
    }

    // Click Listeners
    @Override
    public void onClick(View view) {
        // Previous
        if (view.equals(previous)) {
            week_active--;
            if (week_active < 0) week_active = 0;

            adapter = new WeekAdapter(this, week_active);
            tasks.setAdapter(adapter);

            setTitle();
        }

        // Next
        else if (view.equals(next)) {
            week_active++;

            adapter = new WeekAdapter(this, week_active);
            tasks.setAdapter(adapter);

            setTitle();
        }
    }

    // Task Changed Listener
    @Override
    public void onTaskChanged() {
        onResume();
    }

    // Private Class Methods
    private void setTitle() {
        Calendar start = DateManager.getWeek(week_active).getStart();
        Calendar end = DateManager.getWeek(week_active).getEnd();

        setTitle(String.valueOf(start.get(Calendar.MONTH) + 1) + "/" +
                String.valueOf(start.get(Calendar.DAY_OF_MONTH)) + " - " +
                String.valueOf(end.get(Calendar.MONTH) + 1) + "/" +
                String.valueOf(end.get(Calendar.DAY_OF_MONTH)));
    }

    private void save() {
        // Save
        TaskManager.save();
        TagManager.save();
        Settings.save();
    }
}

File: app/src/main/java/jgappsandgames/smartreminderslite/sort/date/WeekActivity.kt
package jgappsandgames.smartreminderslite.sort.date

// Java
import java.util.Calendar

// Android OS
import android.app.Activity
import android.content.Intent
import android.os.Bundle

// View
import android.view.Menu
import android.view.MenuItem

// App
import jgappsandgames.smartreminderslite.R
import jgappsandgames.smartreminderslite.adapter.TaskAdapter

// Save
import jgappsandgames.smartreminderssave.MasterManager
import jgappsandgames.smartreminderssave.date.DateManager
import jgappsandgames.smartreminderssave.utility.FileUtility

// KotlinX
import kotlinx.android.synthetic.main.activity_date.date_next
import kotlinx.android.synthetic.main.activity_date.date_previous
import kotlinx.android.synthetic.main.activity_date.date_tasks

// App
import jgappsandgames.smartreminderslite.home.FirstRun
import jgappsandgames.smartreminderslite.utility.OptionsUtility

/**
 * WeekActivity
 * Created by joshua on 1/19/2018.
 */
class WeekActivity: Activity(), TaskAdapter.OnTaskChangedListener {
    // Data ----------------------------------------------------------------------------------------
    private var weekActive: Int = 0

    // LifeCycle Methods ---------------------------------------------------------------------------
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_date)

        // First Run
        FileUtility.loadFilePaths(this)
        if (FileUtility.isFirstRun()) startActivity(Intent(this, FirstRun::class.java))
        else MasterManager.load()
        weekActive = 0

        // Set Title
        setTitle()

        // Set Click Listeners
        date_previous.setOnClickListener {
            weekActive--
            if (weekActive < 0) weekActive = 0
            date_tasks.adapter = TaskAdapter(this, this, TaskAdapter.swapTasks(DateManager.getWeekTasks(weekActive)), "")
            setTitle()
        }

        date_next.setOnClickListener {
            weekActive++
            date_tasks.adapter = TaskAdapter(this, this, TaskAdapter.swapTasks(DateManager.getWeekTasks(weekActive)), "")
            setTitle()
        }
    }

    override fun onResume() {
        super.onResume()

        DateManager.create()
        date_tasks.adapter = TaskAdapter(this, this, TaskAdapter.swapTasks(DateManager.getWeekTasks(weekActive)), "")
    }

    override fun onPause() {
        super.onPause()

        save()
    }

    // Menu ----------------------------------------------------------------------------------------
    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_auxilary, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return OptionsUtility.onOptionsItemSelected(this, item, object: OptionsUtility.Save {
            override fun save() {
                this@WeekActivity.save()
            }
        })
    }

    // Task Changed Listener -----------------------------------------------------------------------
    override fun onTaskChanged() {
        onResume()
    }

    // Private Class Methods -----------------------------------------------------------------------
    private fun setTitle() {
        val start = DateManager.getWeek(weekActive).getStart()
        val end = DateManager.getWeek(weekActive).getEnd()

        title = (start.get(Calendar.MONTH) + 1).toString() + "/" +
                start.get(Calendar.DAY_OF_MONTH).toString() + " - " +
                (end.get(Calendar.MONTH) + 1).toString() + "/" +
                end.get(Calendar.DAY_OF_MONTH).toString()
    }

    fun save() {
        MasterManager.save()
    }
}